plugins/outputs/groundwork/slog/slog.go

```diff
@@ -0,0 +1,111 @@
+package slog
+
+import (
+	"context"
+	"encoding/json"
+	"log/slog"
+	"strings"
+	"sync"
+
+	"github.com/influxdata/telegraf"
+)
+
+// NewLogger creates telegraf.Logger adapter for slog.Logger
+func NewLogger(l telegraf.Logger) *slog.Logger {
+	return slog.New(&TlgHandler{Log: l})
+}
+
+// TlgHandler translates slog.Record into telegraf.Logger call
+// inspired by https://github.com/golang/example/blob/master/slog-handler-guide/README.md
+type TlgHandler struct {
+	attrs  []slog.Attr
+	groups []string
+
+	once sync.Once
+
+	GroupsFieldName  string
+	MessageFieldName string
+
+	Log telegraf.Logger
+}
+
+func (h *TlgHandler) Enabled(_ context.Context, level slog.Level) bool {
+	l := h.Log.Level()
+	switch level {
+	case slog.LevelDebug:
+		return l >= telegraf.Debug
+	case slog.LevelInfo:
+		return l >= telegraf.Info
+	case slog.LevelWarn:
+		return l >= telegraf.Warn
+	case slog.LevelError:
+		return l >= telegraf.Error
+	default:
+		return l >= telegraf.Info
+	}
+}
+
+func (h *TlgHandler) Handle(_ context.Context, r slog.Record) error {
+	h.once.Do(func() {
+		if h.GroupsFieldName == "" {
+			h.GroupsFieldName = "logger"
+		}
+		if h.MessageFieldName == "" {
+			h.MessageFieldName = "message"
+		}
+	})
+
+	attrs := make([]slog.Attr, 0, 2+len(h.attrs)+r.NumAttrs())
+	attrs = append(attrs,
+		slog.String(h.MessageFieldName, r.Message),
+		slog.String(h.GroupsFieldName, strings.Join(h.groups, ",")),
+	)
+	for _, attr := range h.attrs {
+		if v, ok := attr.Value.Any().(json.RawMessage); ok {
+			attrs = append(attrs, slog.String(attr.Key, string(v)))
+			continue
+		}
+		attrs = append(attrs, attr)
+	}
+	r.Attrs(func(attr slog.Attr) bool {
+		if v, ok := attr.Value.Any().(json.RawMessage); ok {
+			attrs = append(attrs, slog.String(attr.Key, string(v)))
+			return true
+		}
+		attrs = append(attrs, attr)
+		return true
+	})
```

Comment:
Why do you need to construct an array of attributes instead of directly constructing the message for logging?

Reply:
Here we just preparing log data: adjusting datatype for json.RawMessage for better formatting. Then we letting actual logger make his work.
In output we getting well-structured message.

Reply:
Yeah I was wondering why you don't directly format it into `[]string` instead of keeping them as `[]slog.Attr`...

Reply:
There are non-string Attrs. Convert them here to string, then actual writer will convert it to bytes.. does it make sense? Onetime conversion in writer should be better.

Reply:
I agree.